docs/src/references/synchronize-daemons.md

```diff
@@ -0,0 +1,116 @@
+-   This will be a hard-coded dependency of the name of the plugin and can’t be updated on the fly if someone wants to change the name of the daemon
+    or wants to add a new daemon.
+
+## Proposal 2:  Introduce an `init message` mechanism
+
+Make all the tedge components publish a `retained init` message describing their role (software management, config management, etc)
+ and other details specific to that role.
+
+The proposed topic will be `tedge/<device-id>/init/<cloud>`.
+Here `<cloud>` could be c8y, azure, aws, etc
+
+The message template is {Operation_name: <content>, Operation_name:<content>, Config_types: type1..typen}.
+Here `Operation_name` is the name of the operation to be published to the cloud and 
+the `content` is the content of the operation file, if any.
+Here `Config_types` is the configuration types that are supported by that particular daemon.
+
+Example 1:
+- When the tedge-mapper-c8y is started, it will subscribe to `tedge/<device-id>/init/#` topic
+- When tedge-agent is started, it will publish `{}` an empty `init` message on to `tedge/init/c8y/tedge-agent`
+- Now the mapper picks up this message and creates the `c8y_Restart, c8y_SoftwareUpdat` supported operations file in `/etc/tedge/operations/c8y`. 
+- Mapper sends the updated supported operations list to the c8y cloud.
+- Also, now that the tedge-agent is up and running, the tedge-mapper-c8y will publish the request to get the
+ software list on `tedge/commands/req/software/list`.
+- Now the tedge-agent processes this request and sends the software list to the mapper, once the mapper gets the response it will forward it to the c8y cloud.
+
+Example 2,
+- When tedge-mapper-c8y is started, it will subscribe to `tedge/<device-id>/init/#` topic
+- When c8y-log-plugin is started, it will publish {"LogfileRequest":"", "LogTypes": "software-management, mosquitto-log"} on to `tedge/thin-edge/init` topic
+- Now the mapper picks up this message and creates the c8y_LogfileRequest operation file in `/etc/tedge/operations/c8y`.
+- Also, the mapper creates the `c8y-log-plugin.toml` file, which contains the `supported log types` in `/etc/tedge/c8y`.
+- Mapper sends the updated supported operations list to the c8y cloud. Also, it sends the supported `log types`(118,software-management, mosquitto-log) to the c8y cloud.
+
+> Note: The mapper can also subscribe to cloud-specific init topics like `tedge/<device-id>/init/c8y`, to get a init message from a cloud-specific plugin,
+that does not belong to any of the generic operations categories. For example, any custom operation plugin in case of c8y cloud.
+
+Pros:
+-   This will remove the file system dependency when the daemons want to create the supported operation files on the tedge-mapper-c8y file system.
+-	No dependency on mapper, no hard-coded list of plugins
+-	Very much cloud agnostic because the plugins are categorized based on roles not cloud-specific.
+ For example, the `log` plugin can be used by c8y, azure, aws, etc.
```

Comment:
This is why I don't understand why the log plugin sends its init message on a topic that is specific to a cloud
(i.e. `tedge/<device-id>/init/<cloud>`)

Reply:
Yes, in that case, it need not send on a cloud-specific topic.